master/internal/stream/system.go

```diff
@@ -325,6 +325,30 @@ func (ps *PublisherSet) Websocket(
 	}
 }
 
+// Websocket is an Echo websocket endpoint.
+func (ps *PublisherSet) Websocket(
+	ssupCtx context.Context,
+	socket *websocket.Conn,
+	c echo.Context,
+) error {
+	reqCtx := c.Request().Context()
+	detCtx, ok := c.(*detContext.DetContext)
+	if !ok {
+		log.Errorf("unable to run PublisherSet: expected DetContext but received %t", reflect.TypeOf(c))
+	}
+	user := detCtx.MustGetUser()
+	return ps.entrypoint(ssupCtx, reqCtx, user, &WrappedWebsocket{Conn: socket}, prepareWebsocket)
+}
+
+func prepareWebsocket(msg stream.PreparableMessage) interface{} {
+	if _, ok := msg.(stream.UpsertMsg); ok {
+		return prepareMessageWithCache(msg, nil)
```

Comment:
wait this is a second layer of caching, I think this is unneeded.

I think caching is only ever effectively used for live-streaming messages coming through the publisher, and the caching in the Event is used there.

I think per-message caching is then completely useless.

Reply:
I agree, I'll rewrite the `prepareMessageWithCache` function to be cacheless.